packages/teuchos/core/src/Teuchos_RCP.hpp

```diff
@@ -281,6 +281,15 @@ RCP<T>::RCP(const RCP<T>& r_ptr)
 {}
 
 
+template<class T>
+inline
+RCP<T>::RCP(RCP<T>&& r_ptr)
```

Comment:
@bartlettroscoe It looks like the reason you can't just write `RCP(RCP<T>&&) == default;` is because you want to set `r_ptr.ptr_` to `nullptr` after moving.  Is that right?

Reply:
Correct.  Think about what would happen if you used the default move constructor implementation.   I would leave the raw pointer set in the input RCP `r_ptr` which could have disastrous consequences  if someone were to try to use it.

Reply:
True, though the only requirement for the result of a move is that it be left in an indeterminate but valid state.  I'm guessing the destructor would be unhappy if the pointer were nonnull, though.